bart/sched/pelt.py

```diff
@@ -0,0 +1,347 @@
+
+        :param task: the task we want to simulate the PELT signal for, by
+                     default is the task used to intialize the Simulator
+                     (when specified)
+        :type  task: Task
+
+        :return: A tuple representing the minimum and maximum value of the PELT
+                 singal once stable.
+
+        """
+
+        # Validate input parameters
+        if not isinstance(task, Task):
+            raise ValueError("Wrong time for task parameter")
+
+        # Compute max value
+        max_util  = (1. - pow(self._geom_y, float(task.run_us)/Simulator._signal_max))
+        max_util /= (1. - pow(self._geom_y, float(task.period_us)/Simulator._signal_max))
+
+        # Compute min value, by decaying the maximum value
+        min_util  = (max_util * pow(self._geom_y, float(task.idle_us)/Simulator._signal_max))
+
+        return PELT_Range(Simulator._signal_max * min_util,
+                         Simulator._signal_max * max_util)
+
+
+    def signal(self, task, start_s=0, end_s=10):
+        """
+        Compute the PELT signal for the specified interval.
+
+        :param task: the task we want to simulate the PELT signal for, by
+                     default is the task used to intialize the Simulator
+                     (when specified)
+        :type  task: Task
+
+        :param start_s: the start time in [s]
+        :type  start_s: float
+
+        :param end_s: the end time in [s]
+        :type  end_s: float
+        """
+
+        # Computed PELT samples
+        samples = []
+
+        # Validate input parameters
+        if not isinstance(task, Task):
+            raise ValueError("Wrong time for task parameter")
+
+        # Intervals (in PELT samples) for the signal to compute
+        start_us = int(1e6 * Simulator._sample_us/1e3 * start_s)
+        end_us   = int(1e6 * Simulator._sample_us/1e3 * end_s)
+        #print "Simulation Time Range : ({}:{}) [us]".format(start_us, end_us)
+
+        # Add initial value to the generated signal
+        util = self.init_util
+        t_us = start_us
+        samples.append((float(t_us)/1e6, 0, False, util))
+
+        # Compute following PELT samples
+        t_us = start_us + Simulator._sample_us
+        while t_us <= end_us:
+
+            # Check if the task was running in the current PELT sample
+            running = task.running(float(t_us/1e3))
+
+            # Assume the task was running for all the current PELT sample
+            active_us = Simulator._sample_us if running else 0
+            util = self._geom_sum(util, active_us)
+
+            # Append PELT samples
+            samples.append((float(t_us)/1e6, t_us/Simulator._sample_us, running, util))
+
+            # Prepare for next sample computation
+            t_us += Simulator._sample_us
+
+        # Create DataFrame from computed samples
+        self._df = _DataFrame(samples, columns=['Time', 'PELT_Interval', 'Running', 'util_avg'])
```

Comment:
Also, I am not sure why are we using `util_avg` should it not be just `signal_value`. PELT can theoretically represent any thing. Some of the things being `load_avg` and `util_avg`